doc/dev/cephadm/cephadm-exporter.rst

```diff
@@ -0,0 +1,306 @@
+JSON format and contain the following fields; crt, key, token and port. The JSON content should be validated, and any
+errors detected passed back to the user during the argument parsing phase (before any changes are done).
+
+
+Additional ceph commands
+------------------------
+::
+
+# ceph cephadm generate-exporter-config
+
+This command will create generate a default configuration consisting of; a self signed certificate, a randomly generated
+32 character token and the default port of 9443 for the REST API.
+::
+
+# ceph cephadm set-exporter-config -i <config.json>
+
+Use a JSON file to define the crt, key, token and port for the REST API. The crt, key and token are validated by
+the mgr/cephadm module prior storing the values in the KV store. Invalid or missing entries should be reported to the 
+user.
+::
+
+# ceph cephadm clear-exporter-config
+
+Clear the current configuration (removes the associated keys from the KV store)
+::
+
+# ceph cephadm get-exporter-config
+
+Show the current exporter configuration, in JSON format
+
+
+.. note:: If the service is already deployed any attempt to change or clear the configuration will
+    be denied. In order to change settings you must remove the service, apply the required configuration
+    and re-apply (``ceph orch apply cephadm-exporter``)
+
+
+
+New Ceph Configuration Keys
+===========================
+The exporter configuration is persisted to the monitor's KV store, with the following keys:
+
+| mgr/cephadm/exporter_config
+| mgr/cephadm/exporter_enabled
+
+
+
+RESTful API
+===========
+The primary goal of the exporter is the provision of metadata from the host to the mgr. This interaction takes
+place over a simple GET interface. Although only the GET method is supported, the API provides multiple URLs to
+provide different views on the metadata that has been gathered.
+
+.. csv-table:: Supported URL endpoints
+    :header: "URL", "Purpose"
+
+    "/v1/metadata", "show all metadata including health of all threads"
+    "/v1/metadata/health", "only report on the health of the data gathering threads"
+    "/v1/metadata/disks", "show the disk output (ceph-volume inventory data)"
+    "/v1/metadata/host", "show host related metadata from the gather-facts command"
+    "/v1/metatdata/daemons", "show the status of all ceph cluster related daemons on the host"
+
+Return Codes
+------------
+The following HTTP return codes are generated by the API
+
+.. csv-table:: Supported HTTP Responses
+    :header: "Status Code", "Meaning"
+
+    "200", "OK"
+    "204", "the thread associated with this request is no longer active, no data is returned"
+    "206", "some threads have stopped, so some content is missing"
+    "401", "request is not authorised - check your token is correct"
+    "404", "URL is malformed, not found"
+    "500", "all threads have stopped - unable to provide any metadata for the host"
```

Comment:
Isn't this already documented by saying we're using a REST API? Sounds duplicated to list standard HTTP codes here.

Reply:
yeah - the codes are standard, as you'd expect...but what I'm documenting here is the meaning wrt how the exporter is working or perhaps not working :)